src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -338,25 +337,14 @@ public unsafe BitArray And(BitArray value)
                 case 0: goto Done;
             }
 
-            uint i = 0;
-
-            ref int left = ref MemoryMarshal.GetArrayDataReference<int>(thisArray);
-            ref int right = ref MemoryMarshal.GetArrayDataReference<int>(valueArray);
+            int i = 0;
 
-            if (Vector256.IsHardwareAccelerated)
-            {
-                for (; i < (uint)count - (Vector256IntCount - 1u); i += Vector256IntCount)
-                {
-                    Vector256<int> result = Vector256.LoadUnsafe(ref left, i) & Vector256.LoadUnsafe(ref right, i);
-                    result.StoreUnsafe(ref left, i);
-                }
-            }
-            else if (Vector128.IsHardwareAccelerated)
+            if (Vector.IsHardwareAccelerated)
```

Comment:
Using `Vector<T>` is likely a non-starter right now.

Not only can it not be used with R2R (crossgen) code today since it is variable sized and therefore forces the code to be jitted, but it also is missing certain APIs that are available to `Vector128/256<T>`. A few of these functions, like `ExtractMostSignificantBits` can't be exposed on `Vector<T>`, others like `LoadUnsafe` could be but aren't today.

There is a design being considered (https://github.com/dotnet/designs/pull/268) where we can extend `Vector<T>` to better work with such scenarios and better enable its usage in other scenarios, but that isn't available at the moment and its not 100% clear what shape that will end up as.

Reply:
For this case in particular, `Vector<T>` would likely be similar in perf if we had `LoadUnsafe` and `StoreUnsafe` APIs available.

However, it would still come with the restriction that it could not participate in R2R (and therefore forces jitting on first use) and it may regress scenarios where the backing data is typically "small".

For example, `Vector<T>` on most modern x64 hardware is equivalent to only having the `Vector256<T>` path. On Arm64, its equivalent to only having the `Vector128<T>` path. For x64, this means that inputs less than 32-bytes (and potentially 64-bytes in the future) will behave "worse" than the equivalent on Arm64 as they'll execute as "scalar" rather than as "vector". Likewise, depending on data layout, alignment, and processor, it may behave "worse" for inputs up to ~256 bytes as well.

As indicated, these are scenarios that are being looked at and considered, but its not something that we can easily do today.